slatedb/src/manifest/store.rs

```diff
@@ -93,39 +98,45 @@ impl FenceableManifest {
         system_clock: Arc<dyn SystemClock>,
     ) -> Result<Self, SlateDBError> {
         let clock = system_clock.clone();
-        let fr = FenceableRecord::init(
+        let fr = FenceableTransactionalObject::init(
             stored_manifest.inner,
             manifest_update_timeout,
             system_clock,
             |m: &Manifest| m.compactor_epoch,
             |m: &mut Manifest, e: u64| m.compactor_epoch = e,
         )
-        .await?;
+        .await
+        .map_err(ErrorBridge)?;
         Ok(Self { inner: fr, clock })
     }
 
-    pub(crate) async fn refresh(&mut self) -> Result<(), SlateDBError> {
-        self.inner.refresh().await
+    pub(crate) async fn refresh(&mut self) -> Result<&Manifest, SlateDBError> {
```

Comment:
Is `&Record<Manifest>` style coming in a later PR? I recall us discussing this templated style a while back.

Reply:
Yep. This PR is just aimed at putting in a transactional object API